src/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -194,30 +195,16 @@ internal Authority GetAuthority(IAccount account)
         }
 
         internal virtual AuthenticationRequestParameters CreateRequestParameters(
-            IAcquireTokenCommonParameters commonParameters,
+            AcquireTokenCommonParameters commonParameters,
```

Comment:
>AcquireTokenCommonParameters [](start = 12, length = 28)

why not interface?

Reply:
As we started moving some of the "common" parameters up the stack since they didn't apply to all scenarios (e.g. LoginHint), the pattern of "have a lot of interfaces and one underlying structure with all of the fields" started to become a bug farm of having to ensure what may or may not be valid.  So I moved to explicit parameter structures so we can have the bag of properties for "commonParameters" and then the request-specific bag of parameters (e.g. interactiveparameters).  The AuthenticationRequestParameters are now just built out of the common parameters.  The goal is that AuthenticationRequestParameters only has common concepts in it (and maybe even goes away in teh face of just AcquireTokenCommonParameters) but that refactor was WAY too impactful across the cache and a few other areas that look at this existing model.  Happy to talk more.

---
In reply to: [249074007](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/798#discussion_r249074007) [](ancestors = 249074007)